11.word_count/word_count.rb

```diff
@@ -0,0 +1,18 @@
+text = <<TEXT
+Ruby is an interpreted, high-level, general-purpose programming language. It was designed with an emphasis on programming productivity and simplicity. In Ruby, everything is an object, including primitive data types. It was developed in the mid-1990s by Yukihiro "Matz" Matsumoto in Japan.
+
+Ruby is dynamically typed and uses garbage collection and just-in-time compilation. It supports multiple programming paradigms, including procedural, object-oriented, and functional programming. According to the creator, Ruby was influenced by Perl, Smalltalk, Eiffel, Ada, BASIC, Java, and Lisp.
+TEXT
+
+words = text.scan(/[A-Za-z0-9-]+/)
```

Comment:
```suggestion
words = text.downcase.scan(/[A-Za-z0-9-]+/)
```

最初からdowncaseしちゃう方が効率が良さそう

あと、`半角英数字`は`\w`で置き換えるのもありです。
（この場合アンダースコアも含まれるようになりますが、まあ問題ないでしょう）

```suggestion
words = text.downcase.scan(/[\w-]+/)
```

Reply:
`\w`を使う形で修正いたしました！
メタ文字の役割についても[リファレンスマニュアル](https://docs.ruby-lang.org/ja/3.1/doc/spec=2fregexp.html)から再確認しようと思いました。